crates/primitives/transactions/src/broadcasted_to_blockifier.rs

```diff
@@ -76,88 +79,89 @@ pub fn broadcasted_declare_v0_to_blockifier(
+                            compiled_class_hash: compiled_class_hash_computed,
+                        };
 
-                (
-                    Some(blockifier::execution::contract_class::ClassInfo::new(
-                        &compiled.to_blockifier_class()?,
-                        tx.contract_class.sierra_program.len(),
-                        tx.contract_class.abi.len(),
-                    )?),
-                    Some(class_hash),
-                    Some(ConvertedClass::Sierra(SierraConvertedClass {
-                        class_hash,
-                        info: class_info,
-                        compiled: Arc::new(compiled),
-                    })),
-                )
-            }
-            starknet_core::types::BroadcastedDeclareTransaction::V3(tx) => {
-                let class_hash = tx.contract_class.class_hash();
-                let flatten_sierra_class: FlattenedSierraClass = (*tx.contract_class).clone().into();
-                let (compiled_class_hash, compiled) = flatten_sierra_class.compile_to_casm()?;
-                if tx.compiled_class_hash != compiled_class_hash {
-                    return Err(BroadcastedToBlockifierError::CompiledClassHashMismatch {
-                        expected: tx.compiled_class_hash,
-                        compilation: compiled_class_hash,
-                    });
+                        (
+                            Some(blockifier::execution::contract_class::ClassInfo::new(
+                                &compiled.to_blockifier_class()?,
+                                contract_class.sierra_program.len(),
+                                contract_class.abi.as_ref().map(|abi| abi.len()).unwrap_or(0),
+                            )?),
+                            Some(class_hash),
+                            Some(ConvertedClass::Sierra(SierraConvertedClass {
+                                class_hash,
+                                info: class_info,
+                                compiled: Arc::new(compiled),
+                            })),
+                        )
+                    }
                 }
-                let class_info =
-                    SierraClassInfo { contract_class: Arc::new(flatten_sierra_class), compiled_class_hash };
-
-                (
-                    Some(blockifier::execution::contract_class::ClassInfo::new(
-                        &compiled.to_blockifier_class()?,
-                        tx.contract_class.sierra_program.len(),
-                        tx.contract_class.abi.len(),
-                    )?),
-                    Some(class_hash),
-                    Some(ConvertedClass::Sierra(SierraConvertedClass {
-                        class_hash,
-                        info: class_info,
-                        compiled: Arc::new(compiled),
-                    })),
-                )
             }
-        },
-        _ => (None, None, None),
-    };
+            _ => (None, None, None),
+        };
```

Comment:
Consider refactoring this into sub-functions to make it more readable, I'm havin a hard time sifting through this on review :face_with_spiral_eyes: